### PR TITLE
[#304] fixed time format for slots page

### PR DESCRIPTION
**Pull request by Pavliuk13**
https://user-images.githubusercontent.com/59260347/189676839-04570d7a-caec-4583-b783-f984d7f6184d.mp4

### CODE REVIEW

**Review by hekeemje — CHANGES_REQUESTED** [review]
image
maybe you can do width of input fields more? because there word "AM" and "PM" and there no space to show full text of AM or PM

**Comment by Pavliuk13** [comment]
> image maybe you can do width of input fields more? because there word "AM" and "PM" and there no space to show full text of AM or PM

image

**Review by hekeemje — APPROVED** [review]
looks good

File: frontend/src/app/modules/availability/new-slot/schedule-list-item/schedule-list-item.component.ts [hunk 1]
```diff
@@ -54,12 +56,16 @@ export class ScheduleListItemComponent extends BaseComponent implements OnInit {
         this.itemChange.emit();
     }
 
+    isTwelveHoursFormat() {
+        const user = JSON.parse(localStorage.getItem('user')!) as ILocalUser;
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/new-slot/schedule-list-item/schedule-list-item.component.ts]: You cannot use localStorage in components

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/availability/new-slot/schedule-list-item/schedule-list-item.component.ts (hunk 1 above)
Comment (Pavliuk13) [reply frontend/src/app/modules/availability/new-slot/schedule-list-item/schedule-list-item.component.ts]: I created method in UserService that gets local user from localStorage. Then I used it in SlotDefinitionComponent for getting time format of current user.

File: frontend/src/app/core/services/user.service.ts [hunk 2]
```diff
@@ -112,6 +112,10 @@ export class UserService {
         }
     }
 
+    public getLocalUser(): ILocalUser {
+        return JSON.parse(localStorage.getItem('user')!) as ILocalUser;
```
Comment (tatianahutii) [inline frontend/src/app/core/services/user.service.ts]: You need to add check fo null

File: frontend/src/app/modules/availability/new-slot/schedule-definition/schedule-definition.component.ts [hunk 3]
```diff
@@ -21,6 +23,17 @@ export class ScheduleDefinitionComponent {
 
     displayDays: string[] = getDisplayDays();
 
+    timeFormat?: TimeFormat;
+
+    // eslint-disable-next-line no-empty-function
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/new-slot/schedule-definition/schedule-definition.component.ts]: Do not use eslint-disable-next-line

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/availability/new-slot/schedule-definition/schedule-definition.component.ts [hunk 4]
```diff
@@ -28,11 +30,19 @@ export class ScheduleDefinitionComponent extends BaseComponent {
...
     displayDays: string[] = getDisplayDays();
 
+    timeFormat?: TimeFormat;
+
+    ngOnInit(): void {
+        const user = this.userService.getLocalUser();
+
+        this.timeFormat = user ? user.timeFormat : TimeFormat.TwentyFourHour;
+    }
+
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/new-slot/schedule-definition/schedule-definition.component.ts]: methods need to be after constructor

Comment (tatianahutii) [reply frontend/src/app/modules/availability/new-slot/schedule-definition/schedule-definition.component.ts]: we need to subscribe to event and not load user by yourself

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Review by tatianahutii — APPROVED** [review]

File: frontend/src/app/core/services/user.service.ts [hunk 5]
```diff
@@ -20,6 +20,10 @@ export class UserService {
 
     public userChangedEvent$ = this.onUserChanged.asObservable();
 
+    private currentLocalUser = new BehaviorSubject<ILocalUser | undefined>(undefined);
```
Comment (tatianahutii) [inline frontend/src/app/core/services/user.service.ts]: What is different beetwen userChangedEvent and currentLocalUserEmitted?

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Review by tatianahutii — APPROVED** [review]

**Review by apofis96 — APPROVED** [review]

**Pull request merged**